Unable to use @ClientExceptionMapper in native build

### Describe the bug

While building I get this exception;

Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: org.acme.RestService.toException(javax.ws.rs.core.Response).

I tried making the method public, and registering the interface for reflection but it did not work.

cc @geoand 

### Expected behavior

Graalvm build should work

### Actual behavior

Graalvm build fails with an exception

### How to Reproduce?

I dont have a reproducer now, but I can provide one in needed.

### Output of `uname -a` or `ver`

Darwin Kernel Version 20.6.0

### Output of `java -version`

openjdk version "17.0.2" 2022-01-18

### GraalVM version (if different from Java)

GraalVM 21.3.1 Java 17 CE

### Quarkus version or git rev

2.7.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

### Additional information

_No response_

I am oddly totally unable to create a reproducer for it, even if I copy my annotation in totality.

I can at least put my full stacktrace here in case there is anything that could shed some light on what is happening, in the meantime I try to create a reproducer. With my "real" app, I get it consistently.


```
[INFO] --- quarkus-maven-plugin:2.7.1.Final:build (default) @ quarkus ---
[WARNING] [io.quarkus.micrometer.deployment.binder.mpmetrics.MicroprofileMetricsProcessor] This application uses the MP Metrics API. The micrometer extension currently provides a compatibility layer that supports the MP Metrics API, but metric names and recorded values will be different. Note that the MP Metrics compatibility layer will move to a different extension in the future.
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /Users/alex/code/quarkus/target/quarkus-1.0.0-native-image-source-jar/quarkus-1.0.0-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /Users/alex/code/quarkus/target/quarkus-1.0.0-native-image-source-jar/quarkus-1.0.0-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on GraalVM 21.3.1 Java 17 CE (Java Version 17.0.2+8-jvmci-21.3-b09)
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildRunner] /Users/alex/tools/graalvm-ce-java17-21.3.1/Contents/Home/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=3 -J-Duser.language=en -J-Duser.country=CA -J-Dfile.encoding=UTF-8 -H:-ParseOnce -J--add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED -J--add-opens=java.base/java.text=ALL-UNNAMED -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -J-Djava.awt.headless=true -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https -H:-UseServiceLoaderFeature -H:+StackTrace quarkus-1.0.0-runner -jar quarkus-1.0.0-runner.jar
[quarkus-1.0.0-runner:6568]    classlist:   2,348.30 ms,  1.19 GB
[quarkus-1.0.0-runner:6568]        (cap):   1,520.81 ms,  1.19 GB
[quarkus-1.0.0-runner:6568]        setup:   3,292.22 ms,  1.19 GB
10:21:06,604 INFO  [org.hib.val.int.uti.Version] HV000001: Hibernate Validator 6.2.1.Final
10:21:27,914 INFO  [org.jbo.threads] JBoss Threads version 3.4.2.Final
[quarkus-1.0.0-runner:6568]     (clinit):     737.70 ms,  4.07 GB
[quarkus-1.0.0-runner:6568]   (typeflow):   3,332.12 ms,  4.07 GB
[quarkus-1.0.0-runner:6568]    (objects):  33,179.56 ms,  4.07 GB
[quarkus-1.0.0-runner:6568]   (features):   2,220.16 ms,  4.07 GB
[quarkus-1.0.0-runner:6568]     analysis:  41,377.00 ms,  4.07 GB
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: org.acme.rest.EventService.toException(javax.ws.rs.core.Response). To diagnose the issue you can use the --allow-incomplete-classpath option. The missing method is then reported at run time when it is accessed the first time.
Detailed message:
Trace:
    at parsing org.acme.rest.EventService$$CDIWrapper.toException(Unknown Source)
Call path from entry point to org.acme.rest.EventService$$CDIWrapper.toException(Response):
    at org.acme.rest.EventService$$CDIWrapper.toException(Unknown Source)
    at org.acme.rest.EventService$$CDIWrapper_Subclass.toException$$superforward1(Unknown Source)
    at org.acme.rest.EventService$$CDIWrapper_Subclass$$function$$3.apply(Unknown Source)
    at sun.security.ec.ParametersMap$1.get(ParametersMap.java:78)
    at com.oracle.svm.core.jdk.SystemPropertiesSupport.initializeLazyValue(SystemPropertiesSupport.java:216)
    at com.oracle.svm.core.jdk.SystemPropertiesSupport.getProperty(SystemPropertiesSupport.java:169)
    at com.oracle.svm.core.jdk.Target_java_lang_System.getProperty(JavaLangSubstitutions.java:287)
    at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_ARRAY_System_getProperty_deeeaa72a006d330408a3b7d002c7533e108bc28(generated:0)

com.oracle.svm.core.util.UserError$UserException: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: org.acme.rest.EventService.toException(javax.ws.rs.core.Response). To diagnose the issue you can use the --allow-incomplete-classpath option. The missing method is then reported at run time when it is accessed the first time.
Detailed message:
Trace:
    at parsing org.acme.rest.EventService$$CDIWrapper.toException(Unknown Source)
Call path from entry point to org.acme.rest.EventService$$CDIWrapper.toException(Response):
    at org.acme.rest.EventService$$CDIWrapper.toException(Unknown Source)
    at org.acme.rest.EventService$$CDIWrapper_Subclass.toException$$superforward1(Unknown Source)
    at org.acme.rest.EventService$$CDIWrapper_Subclass$$function$$3.apply(Unknown Source)
    at sun.security.ec.ParametersMap$1.get(ParametersMap.java:78)
    at com.oracle.svm.core.jdk.SystemPropertiesSupport.initializeLazyValue(SystemPropertiesSupport.java:216)
    at com.oracle.svm.core.jdk.SystemPropertiesSupport.getProperty(SystemPropertiesSupport.java:169)
    at com.oracle.svm.core.jdk.Target_java_lang_System.getProperty(JavaLangSubstitutions.java:287)
    at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_ARRAY_System_getProperty_deeeaa72a006d330408a3b7d002c7533e108bc28(generated:0)

    at com.oracle.svm.core.util.UserError.abort(UserError.java:87)
    at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:233)
    at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:759)
    at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:529)
    at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:488)
    at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:403)
    at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:569)
    at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:122)
    at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:599)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: org.acme.rest.EventService.toException(javax.ws.rs.core.Response). To diagnose the issue you can use the --allow-incomplete-classpath option. The missing method is then reported at run time when it is accessed the first time.
Detailed message:
Trace:
    at parsing org.acme.rest.EventService$$CDIWrapper.toException(Unknown Source)
Call path from entry point to org.acme.rest.EventService$$CDIWrapper.toException(Response):
    at org.acme.rest.EventService$$CDIWrapper.toException(Unknown Source)
    at org.acme.rest.EventService$$CDIWrapper_Subclass.toException$$superforward1(Unknown Source)
    at org.acme.rest.EventService$$CDIWrapper_Subclass$$function$$3.apply(Unknown Source)
    at sun.security.ec.ParametersMap$1.get(ParametersMap.java:78)
    at com.oracle.svm.core.jdk.SystemPropertiesSupport.initializeLazyValue(SystemPropertiesSupport.java:216)
    at com.oracle.svm.core.jdk.SystemPropertiesSupport.getProperty(SystemPropertiesSupport.java:169)
    at com.oracle.svm.core.jdk.Target_java_lang_System.getProperty(JavaLangSubstitutions.java:287)
    at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_ARRAY_System_getProperty_deeeaa72a006d330408a3b7d002c7533e108bc28(generated:0)

    at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:126)
    at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:756)
    ... 6 more
Caused by: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: org.acme.rest.EventService.toException(javax.ws.rs.core.Response). To diagnose the issue you can use the --allow-incomplete-classpath option. The missing method is then reported at run time when it is accessed the first time.
    at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.reportUnresolvedElement(SharedGraphBuilderPhase.java:307)
    at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.handleUnresolvedMethod(SharedGraphBuilderPhase.java:298)
    at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.handleUnresolvedInvoke(SharedGraphBuilderPhase.java:252)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genInvokeSpecial(BytecodeParser.java:1805)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genInvokeSpecial(BytecodeParser.java:1795)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5418)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3477)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3437)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3282)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1145)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1030)
    at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
    at com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:81)
    at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
    at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:212)
    at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
    at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
    at com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:132)
    at com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:616)
    at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:163)
    at com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:321)
    at com.oracle.graal.pointsto.flow.MethodTypeFlow.createTypeFlow(MethodTypeFlow.java:293)
    at com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureTypeFlowCreated(MethodTypeFlow.java:282)
    at com.oracle.graal.pointsto.flow.MethodTypeFlow.addContext(MethodTypeFlow.java:103)
    at com.oracle.graal.pointsto.DefaultAnalysisPolicy$DefaultSpecialInvokeTypeFlow.onObservedUpdate(DefaultAnalysisPolicy.java:363)
    at com.oracle.graal.pointsto.flow.TypeFlow$1.run(TypeFlow.java:449)
    at com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:188)
    at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:172)
    at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
    at java.base/java.util.concurrent.ForkJoinPool.externalHelpQuiescePool(ForkJoinPool.java:2104)
    at java.base/java.util.concurrent.ForkJoinPool.awaitQuiescence(ForkJoinPool.java:3321)
    at com.oracle.graal.pointsto.util.CompletionExecutor.complete(CompletionExecutor.java:238)
    at com.oracle.graal.pointsto.PointsToAnalysis.checkObjectGraph(PointsToAnalysis.java:680)
    at com.oracle.graal.pointsto.PointsToAnalysis.finish(PointsToAnalysis.java:644)
    at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:704)
    ... 6 more
[quarkus-1.0.0-runner:6568]      [total]:  47,320.80 ms,  4.07 GB
# Printing build artifacts to: /Users/alex/code/quarkus/target/quarkus-1.0.0-native-image-source-jar/quarkus-1.0.0-runner.build_artifacts.txt
Error: Image build request failed with exit status 1
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  58.411 s
[INFO] Finished at: 2022-02-17T10:21:47-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.7.1.Final:build (default) on project quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]     [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: io.quarkus.deployment.pkg.steps.NativeImageBuildStep$ImageGenerationFailureException: Image generation failed. Exit code: 1
[ERROR]     at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.imageGenerationFailed(NativeImageBuildStep.java:391)
[ERROR]     at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.build(NativeImageBuildStep.java:232)
[ERROR]     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]     at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]     at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)
[ERROR]     at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]     at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]     at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR]     at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

@geoand  I have made a reproducer [here](https://github.com/manofthepeace/quarkus-reproducer-mapperinterceptor)

I see a couple of issues. So in the end what I thought was working wasn't. Build would work, but it would not work at runtime

1. In EventService.java you will see I commented out the `@NoClassInterceptors`
in this scenario, `mvn clean package -Pnative` will fail with the exception I pasted above

2. In EventService.java if you uncomment the `@NoClassInterceptor`
the native build will go through, but will fail right away at runtime with the following;
```
Caused by: java.lang.IllegalArgumentException: Failed to instantiate provider class org.acme.restClient.Event.EventService_toException_ResponseExceptionMapper_755d4bf7b8544dc0f7351820572df6985c3bde0f. Does it have a public no-arg constructor?
...
Caused by: java.lang.NoSuchMethodException: org.acme.restClient.Event.EventService_toException_ResponseExceptionMapper_755d4bf7b8544dc0f7351820572df6985c3bde0f.<init>()
``` 

Do you  think maybe the bug could  be reopened?

--- a/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java
+++ b/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java
@@ -400,7 +400,7 @@ class RestClientReactiveProcessor {
                 // but emitting the correct invokespecial instruction would become convoluted
                 // (as invokespecial may only reference a method from a _direct_ super interface)
                 for (MethodInfo method : jaxrsInterface.methods()) {
-                    boolean isDefault = !Modifier.isAbstract(method.flags());
+                    boolean isDefault = !Modifier.isAbstract(method.flags()) && !Modifier.isStatic(method.flags());
                     if (isDefault) {
                         methodsToImplement.add(method);
                     }

